### PR TITLE
(maint) Adding MIGRATION.md for migrating from voxpupuli to puppetlabs

### PR DESCRIPTION
**Pull request by hunner**

### CODE REVIEW

**Comment by hunner** [comment]
@jbondpdx It would be great if you could give this a "this is terrible and shouldn't be merged" kind of review! It's not going to be a long-living reference document.

**Review by jpogran — APPROVED** [review]

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+apppool_idle_timeout_action      | idle_timeout_action
+
+### `iis_site`
+This type is still called `iis_site`
+
+The sslFlags attribute was added to specify the binding used for Secure Sockets
+Layer certificates.
+https://www.iis.net/configreference/system.applicationhost/sites/site/bindings/binding
+
+sslFlags 	Optional uint attribute.
```
Comment (jbondpdx) [inline MIGRATION.md]: typo in unit

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+With a Server Name Indicator (SNI), the host name is exchanged as part of the
+SSL handshake. SNI is enabled in the Add Site Binding dialog box when you add a
+binding with a type of HTTPS. This is especially useful for SSL connections
+that host multiple servers on a single network address. For more information,
+see IIS 8.0 Server Name Indication (SNI): SSL Scalability.
+
+The sslFlags attribute is only set when the protocol is https.
+
+The default value is 0.
+#### Parameter mapping:
```
Comment (jbondpdx) [inline MIGRATION.md]: should have a line space before (and after, really) headers

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+name        | identical
+path        | physicalpath
+app_pool    | applicationpool
+host_header | bindings hostname*
+protocol    | bindings protocol*
+ip          | bindings bindinginformation*
+port        | bindings bindinginformation*
+ssl         | bindings sslflags*
+
+\* The `bindings` parameter on the new `iis_site` take an array of one or more bindings as a hash and have the following keys per hash:
```
Comment (jbondpdx) [inline MIGRATION.md]: verb agreement; change take -> takes and have -> has

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+- protocol
+- bindinginformation
+- sslflags
+
+The bindinginformation is in the form `<ip>:<port>`
```
Comment (jbondpdx) [inline MIGRATION.md]: Suggest changing "bindinginformation" to  "bindinginformation key" so it doesn't look so much like we accidentally ran the two words together.

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+### `iis::manage_binding`
+All of these properties are managed directly by the `iis_site::bindings` property and are not declared separately, as the `iis_site` and `iis::manage_binding` resources are. The following describe the `iis_site::bindings` hash keys for the mapping:
```
Comment (jbondpdx) [inline MIGRATION.md]: Verb agreement problem in the last sentence; for better readability, I suggest changing to:

The `iis_site::bindings` hash keys for the map are as follows:

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+- application deployment
+- common http
+- health and diagnostics
+- management tools
+- performance
+- security
+
+
+### `iis_virtualdirectory`
+MISSING! This will be added with the proposed changes:
```
Comment (jbondpdx) [inline MIGRATION.md]: Is it the parameter mapping that are the proposed changes? When and how do are those changes happening? I'm confused by this one.

**Review by jbondpdx — CHANGES_REQUESTED** [review]
In addition to the line comments, there should be line spaces before and after all headers, and no hard line breaks within paragraphs.

**Review by jbondpdx — APPROVED** [review]

**Pull request merged**